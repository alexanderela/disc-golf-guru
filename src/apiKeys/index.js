export const weatherAPIKey = 'a1b5ca4a4ec033f354008b4971f71994'