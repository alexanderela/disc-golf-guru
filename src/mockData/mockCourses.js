const mockCourses = [
  {
    course_id: '4712',
    name: 'Genesee Valley Park',
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zipcode: '14620',
    latitude: '43.119428',
    longitude: '-77.636679',
    street_addr: '99 Elmwood Ave.',
    reviews: '7',
    rating: '3.29',
    private: '0',
    paytoplay: '0',
    tee_1_clr: 'FF0000',
    tee_2_clr: '',
    tee_3_clr: '',
    tee_4_clr: '',
    dgcr_url: 'http://www.dgcoursereview.com/course.php?id=4712',
    dgcr_mobile_url: 'http://www.dgcoursereview.com/mobile/course.php?id=4712',
    rating_img: 'http://www.dgcoursereview.com/images/rating/discs_3.5.png',
    rating_img_small:
      'http://www.dgcoursereview.com/images/rating/discs_sm_3.5.png',
  },
  {
    course_id: '7628',
    name: 'Lookup Park',
    holes: '9',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zipcode: '14467',
    latitude: '43.045582',
    longitude: '-77.617742',
    street_addr: '3850 East Henrietta Rd.',
    reviews: '4',
    rating: '3.13',
    private: '0',
    paytoplay: '0',
    tee_1_clr: 'FF0000',
    tee_2_clr: '',
    tee_3_clr: '',
    tee_4_clr: '',
    dgcr_url: 'http://www.dgcoursereview.com/course.php?id=7628',
    dgcr_mobile_url: 'http://www.dgcoursereview.com/mobile/course.php?id=7628',
    rating_img: 'http://www.dgcoursereview.com/images/rating/discs_3.png',
    rating_img_small:
      'http://www.dgcoursereview.com/images/rating/discs_sm_3.png',
  },
  {
    course_id: '4937',
    name: 'Monroe Community College DGC',
    holes: '18',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zipcode: '14623',
    latitude: '43.103033',
    longitude: '-77.605526',
    street_addr: '',
    reviews: '16',
    rating: '2.94',
    private: '0',
    paytoplay: '0',
    tee_1_clr: 'FFFF00',
    tee_2_clr: '000000',
    tee_3_clr: '',
    tee_4_clr: '',
    dgcr_url: 'http://www.dgcoursereview.com/course.php?id=4937',
    dgcr_mobile_url: 'http://www.dgcoursereview.com/mobile/course.php?id=4937',
    rating_img: 'http://www.dgcoursereview.com/images/rating/discs_3.png',
    rating_img_small:
      'http://www.dgcoursereview.com/images/rating/discs_sm_3.png',
  },
  {
    course_id: '6795',
    name: 'Ellison Park',
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zipcode: '14610',
    latitude: '43.153154',
    longitude: '-77.527428',
    street_addr: '',
    reviews: '12',
    rating: '3.54',
    private: '0',
    paytoplay: '0',
    tee_1_clr: '',
    tee_2_clr: '',
    tee_3_clr: '0000CC',
    tee_4_clr: '',
    dgcr_url: 'http://www.dgcoursereview.com/course.php?id=6795',
    dgcr_mobile_url: 'http://www.dgcoursereview.com/mobile/course.php?id=6795',
    rating_img: 'http://www.dgcoursereview.com/images/rating/discs_3.5.png',
    rating_img_small:
      'http://www.dgcoursereview.com/images/rating/discs_sm_3.5.png',
  },
  {
    course_id: '6443',
    name: 'North Ponds DGC',
    holes: '9',
    city: 'Webster',
    state: 'NY',
    country: 'United States',
    zipcode: '14580',
    latitude: '43.216217',
    longitude: '-77.439955',
    street_addr: '750 Holt Rd.',
    reviews: '2',
    rating: '1.75',
    private: '0',
    paytoplay: '0',
    tee_1_clr: '',
    tee_2_clr: 'FF9900',
    tee_3_clr: '',
    tee_4_clr: '',
    dgcr_url: 'http://www.dgcoursereview.com/course.php?id=6443',
    dgcr_mobile_url: 'http://www.dgcoursereview.com/mobile/course.php?id=6443',
    rating_img: 'http://www.dgcoursereview.com/images/rating/discs_2.png',
    rating_img_small:
      'http://www.dgcoursereview.com/images/rating/discs_sm_2.png',
  },
];

export default mockCourses;
