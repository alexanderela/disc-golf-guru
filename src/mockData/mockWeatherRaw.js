const mockWeather = {
    "coord": {
        "lon": -77.48,
        "lat": 43.13
    },
    "weather": [
        {
            "id": 800,
            "main": "Clear",
            "description": "clear sky",
            "icon": "01n"
        }
    ],
    "base": "stations",
    "main": {
        "temp": 42.08,
        "pressure": 1027,
        "humidity": 76,
        "temp_min": 41,
        "temp_max": 44.06
    },
    "visibility": 16093,
    "wind": {
        "speed": 6.93,
        "deg": 140
    },
    "clouds": {
        "all": 1
    },
    "dt": 1541388900,
    "sys": {
        "type": 1,
        "id": 2143,
        "message": 0.0043,
        "country": "US",
        "sunrise": 1541418600,
        "sunset": 1541454985
    },
    "id": 420026615,
    "name": "Rochester",
    "cod": 200
}

export default mockWeather