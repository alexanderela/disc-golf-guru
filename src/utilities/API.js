export const fetchData = async url => {
  const response = await fetch(url);
  const responseJson = await response.json();
  return responseJson;
};
