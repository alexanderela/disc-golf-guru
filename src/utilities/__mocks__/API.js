export const fetchData = jest.fn().mockImplementation(() => {
  return [];
});
