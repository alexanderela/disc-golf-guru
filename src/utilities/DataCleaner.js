import * as API from './API.js'
import * as APIKey from '../apiKeys'

export const fetchGolfCoursesByZip = async (enteredZipCode) => {
	const url = `https://www.dgcoursereview.com/api_test/?key=${APIKey.discGolfKey}&mode=findzip&zip=${enteredZipCode}&rad=10&sig=${APIKey.discGolfSig}`
	const golfCourseData = await API.fetchData(url)
	const golfCourseResults = await returnGolfCourseData(golfCourseData)
	// console.log(golfCourseResults)
	return golfCourseResults
}

export const returnGolfCourseData = async (golfCourses) => {
	const golfCoursePromises = golfCourses.map( async course => {
		return {
			id: course.course_id,
			name: course.name,
			holes: course.holes,
			city: course.city,
			state: course.state,
			country: course.country,
			zipCode: course.zipcode,
			street_addr: course.street_addr,
			reviews: course.reviews,
			rating: course.rating,
			payToPlay: course.paytoplay,
			// reviewsLink: course.dgcr_url,
		}
	})
	return Promise.all(golfCoursePromises)
}


export const fetchWeather = async () => {

}

export const returnWeatherData = async () => {

}



