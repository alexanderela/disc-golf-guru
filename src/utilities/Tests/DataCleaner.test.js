import * as DataCleaner from '../DataCleaner';
import * as API from '../API';
import * as APIKey from '../../apiKeys'
import mockCourses from '../../mockData/mockCourses.js';
import mockWeatherRaw from '../../mockData/mockWeatherRaw.js';
jest.mock('../API');

describe('DataCleaner', () => {
		let url;
		let input;
		let expected;
		let result;
		
	describe('fetchGolfCourseData', () => {
		beforeEach(() => {
			input = 14526;
		})

		it('should call fetchData with the correct parameters if passed valid zip', async () => {
			url = `https://www.dgcoursereview.com/api_test/?key=${APIKey.discGolfKey}&mode=findzip&zip=${input}&rad=10&sig=${APIKey.discGolfSig}`;
			DataCleaner.fetchGolfCourseData(input);
			expect(API.fetchData).toHaveBeenCalledWith(url)
		})

		it('should call fetchData with the correct parameters if passed an invalid zip', async () => {
			input = 'rochester'
			url = `https://www.dgcoursereview.com/api_test/?key=${APIKey.discGolfKey}&mode=findloc&city=${input}&rad=10&state=NY&country=US&sig=${APIKey.discGolfSig}`;
			DataCleaner.fetchGolfCourseData(input);
			expect(API.fetchData).toHaveBeenCalledWith(url)
		})

		it('should return a resolved array', async () => {
			result = await DataCleaner.fetchGolfCourseData(input);
			expect(result).toEqual([]);
		})
	})

	describe('formatGolfCourseData', () => {
		it('should format golf course objects', async () => {
			result = DataCleaner.formatGolfCourseData(mockCourses);
			expect(result).toMatchSnapshot();
		})
	})

	describe('fetchWeatherData', () => {
		beforeEach(() => {
			input = 14526;
		})

		it('should call fetchData with the correct parameters if passed a valid zip', async () => {
			url = `https://cors-anywhere.herokuapp.com/api.openweathermap.org/data/2.5/weather?zip=${input}&units=imperial&APPID=${APIKey.weatherKey}`;
			DataCleaner.fetchWeatherData(input);
			expect(API.fetchData).toHaveBeenCalledWith(url)

		})

		it('should call fetchData with the correct parameters if passed an invalid zip', async () => {
			input = 'rochester'
			url = `https://cors-anywhere.herokuapp.com/api.openweathermap.org/data/2.5/weather?q=${input}&units=imperial&APPID=${APIKey.weatherKey}`
			DataCleaner.fetchWeatherData(input);
			expect(API.fetchData).toHaveBeenCalledWith(url)
		})

		it('should return a resolved object', async () => {
			expected = {
				"description": "Clear sky", 
				"humidity": 76, 
				"icon": "01n", 
				"id": 420026615, 
				"temp": 42.08, 
				"wind": 6.93
			}
			result = await DataCleaner.formatWeatherData(mockWeatherRaw);
			expect(result).toEqual(expected);
		})
	})

	describe('formatWeatherData', () => {
		it('should format weather data', async () => {
			result = DataCleaner.formatWeatherData(mockWeatherRaw);
			expect(result).toMatchSnapshot()
		})
	})

	describe('convertNumberToBoolean', () => {
		it('should convert 0 to the word No', () => {
			const expected = 'No'
			const result = DataCleaner.convertNumberToBoolean('0')
			expect(result).toEqual(expected)
		})

		it('should convert any number except 0 to the word Yes', () => {
			const expected = 'Yes'
			const result = DataCleaner.convertNumberToBoolean('3')
			expect(result).toEqual(expected)			
		})
	})

	describe('validateZip', () => {
		it('should return true if number is a valid zip', () => {
			const result = DataCleaner.validateZip(14526)
			expect(result).toBe(true)
		})	

		it('should return false if number is not a valid zip', () => {
			const result = DataCleaner.validateZip(145268)
			expect(result).toBe(false)		
		})	
	})

	describe('capitalizeString', () => {
		it('should capitalize the first letter in a string', () => {
			const expected = 'Butter'
			const result = DataCleaner.capitalizeString('butter')
			expect(result).toEqual(expected)
		})
	})

})