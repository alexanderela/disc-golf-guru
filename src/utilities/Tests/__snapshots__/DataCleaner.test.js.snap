// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCleaner returnGolfCourseData should format golf course objects 1`] = `Promise {}`;
